internal/cmdplan/command.go

```diff
@@ -0,0 +1,285 @@
+	}
+	return nil
+}
+
+func hasChanges(plan *v1alpha1.Plan) bool {
+	for _, a := range plan.Spec.Actions {
+		if a.Type != v1alpha1.Unchanged {
+			return true
+		}
+	}
+	return false
+}
+
+func printEntryWithColor(prefix string, color print.Color, action v1alpha1.Action, ioStreams genericclioptions.IOStreams) {
+	txt := print.SprintfWithColor(color, "%s%s %s/%s %s/%s\n", EntryPrefix, prefix, action.Group, action.Kind, action.Namespace, action.Name)
+	fmt.Fprint(ioStreams.Out, txt)
+}
+
+func printEntry(prefix string, action v1alpha1.Action, ioStreams genericclioptions.IOStreams) {
+	fmt.Fprintf(ioStreams.Out, "%s%s %s/%s %s/%s\n", EntryPrefix, prefix, action.Group, action.Kind, action.Namespace, action.Name)
+}
+
+func findAndPrintDiff(before, after *unstructured.Unstructured, prefix string, ioStreams genericclioptions.IOStreams) {
+	diff, err := diffObjects(before, after)
+	if err != nil {
+		panic(err)
+	}
+	for _, d := range diff {
+		if d.Path == ".metadata.generation" || d.Path == ".metadata.managedFields.0.time" {
+			continue
+		}
+
+		switch d.Type {
+		case "LeftAdd":
+			txt := print.SprintfWithColor(print.RED, "%s-%s: %s\n", prefix, d.Path, strings.TrimSpace(fmt.Sprintf("%v", d.Left)))
+			fmt.Fprint(ioStreams.Out, txt)
+		case "RightAdd":
+			txt := print.SprintfWithColor(print.GREEN, "%s+%s: %s\n", prefix, d.Path, strings.TrimSpace(fmt.Sprintf("%v", d.Right)))
+			fmt.Fprint(ioStreams.Out, txt)
+		case "Change":
+			txt1 := print.SprintfWithColor(print.RED, "%s-%s: %s\n", prefix, d.Path, strings.TrimSpace(fmt.Sprintf("%v", d.Left)))
+			fmt.Fprint(ioStreams.Out, txt1)
+			txt2 := print.SprintfWithColor(print.GREEN, "%s+%s: %s\n", prefix, d.Path, strings.TrimSpace(fmt.Sprintf("%v", d.Right)))
+			fmt.Fprint(ioStreams.Out, txt2)
+		}
+	}
+}
+
+func findResource(objs []*unstructured.Unstructured, group, kind, namespace, name string) (*unstructured.Unstructured, bool) {
+	for i := range objs {
+		o := objs[i]
+		gvk := o.GroupVersionKind()
+		if gvk.Group == group && gvk.Kind == kind && o.GetName() == name && o.GetNamespace() == namespace {
+			return o, true
+		}
+	}
+	return nil, false
+}
+
+func printKRMWithPrefix(u *unstructured.Unstructured, prefix string, ioStreams genericclioptions.IOStreams) {
+	b, err := yaml.Marshal(u.Object)
+	if err != nil {
+		panic(fmt.Errorf("unable to marshal resource: %v", err))
+	}
+	printWithPrefix(string(b), prefix, ioStreams)
+}
+
+func printWithPrefix(text, prefix string, ioStreams genericclioptions.IOStreams) {
+	scanner := bufio.NewScanner(strings.NewReader(text))
+	for scanner.Scan() {
+		fmt.Fprintf(ioStreams.Out, "%s%s\n", prefix, scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		panic(fmt.Errorf("error reading text: %v", err))
+	}
+}
+
+func printKRM(plan *v1alpha1.Plan, ioStreams genericclioptions.IOStreams) error {
+	b, err := yaml.Marshal(plan)
+	if err != nil {
+		return fmt.Errorf("failed to marshall plan into yaml: %w", err)
+	}
+	rn, err := yaml.Parse(string(b))
```

Comment:
What's the difference vs just printing `b`?  (Can we add a comment to explain what we're doing here...)

Reply:
This outputs the resources in the KRM function `ResourceList` resource. I'm not sure if this is the right way to do it, so I'll look into it. We do want to be able to output both the plan and the input resources, but that is something I am looking to add later.